packages/front-end/components/DataViz/DataVizFilter.tsx

```diff
@@ -97,222 +84,216 @@ export default function DataVizFilter({
 
     // Create updated filter with new config, maintaining discriminated union structure
     const updatedFilter: FilterConfiguration = {
-      ...currentFilter,
+      ...filter,
       config: newConfig,
     } as FilterConfiguration;
 
-    updateFilter(updatedFilter);
+    onFilterChange(updatedFilter);
   };
 
-  const removeFilter = () => {
-    const newFilters = [...filters];
-    newFilters.splice(filterIndex, 1);
-    onDataVizConfigChange({
-      ...dataVizConfig,
-      filter: newFilters,
-    });
-  };
-
-  const createDefaultFilterForType = (
+  const createFilterConfig = (
     column: string,
-    type: "string" | "number" | "date",
+    type: FilterConfiguration["type"],
+    filterType?: string,
```

Comment:
nit (non-blocking): `filter.type` and `filter.filterType` are very confusing names unfortunately. Maybe we can figure out a more clear naming scheme in the near future?

Reply:
Agreed.

I do think now would be the time to change it.

We currently have

```ts
  z.object({
    column: z.string(),
    type: z.literal("date"), // I think maybe we can change it to columnType
    filterType: z.enum(["today", "last7Days", "last30Days"]), // we could change this to 
    config: z.object({}).optional(), // No config needed
  }),
  ```
  
  My thought is we do something like:
  
  ```ts
  column: z.string(),
  columnType: z.literal("date"),
  filterMethod: z.enum("today", "last7Days", "last30Days"]),
  ```

Reply:
Added these updates [here](https://github.com/growthbook/growthbook/pull/4390/commits/5bc6073b582118b7d502e285ff47a5fe3c3194a0).